### PR TITLE
Move to Python 3

### PR DESCRIPTION
**Pull request by alanhdu**
So far, we've been lucky that Facebook handles all kinds of unicode things for us. Unfortunately, CUIT doesn't (so [web scraping](https://calendar.columbia.edu/sundial/webapi/get.php?vt=detail&id=78066&con=standalone&br=seas_events) can be a pain).

If we have to deal with text encoding issues, I'd much rather move to Python 3. Unicode in Python 2's a nightmare. Besides, Python 3 gives us [all](https://www.python.org/dev/peps/pep-3107/) [kinds](http://mypy-lang.org/) [of](http://asmeurer.github.io/python3-presentation/python3-presentation.pdf) [goodies](http://compiletoi.net/fast-scraping-in-python-with-asyncio.html) we can start playing with.

Anyways, the actual port is fairly straightforward:
- Trivial syntax changes (e.g. `print()` instead of `print`)
- A bit more care with using `decode` and `encode`
- Rip out `facebook-sdk` and write a scraper w/ `requests` (this is actually necessary anyways: `facebook-sdk` still uses the v1.0 API, which will be deprecated on April 30th. Considering it's been two years since a release of `facebook-sdk`, I'm not hopeful about our prospects). This lets us use facebook's bulk-querying apis, which is nice.
- Use a [branch](https://github.com/tonet666p/Flask-WhooshAlchemy/tree/python-3-compatibility) of `flask-whooshalchemy` that's python 3 compatible. There's a PR to merge into upstream, but I doubt that'll happen

Note: to avoid unicode errors, you have to `rm -rf whoosh_alchemy`, `rm events.db`, and run `python create.py` again to rebuild the whoosh indexes.

~~This depends on #80 (to preempt merge conflicts).~~ Naturally, I'd like to merge soon to avoid huge tangles of merge conflicts later.

### CODE REVIEW

**Comment by alanhdu** [comment]
@davidhao3300 @kl2806 @pkathail @ohEmily 

Obviously, this is a fairly major change, so we should talk about it before doing it. I think it's a good idea, but it'd really affect @ohEmily and @kl2806 the most (w/ the scrapers).

~~I haven't yet managed to figure out how to set up Vagrant w/ Python 3.4, so this is still incomplete.~~ (Done)

**Comment by ohEmily** [comment]
not worried about Python version switch. I think we wanted to discuss this in person but I suddenly can't make it.

File: config/bootstrap.sh [hunk 1]
```diff
@@ -14,12 +14,13 @@ apt-get update
 install sqlite3
 
 # install python
-install python
-install python-pip
-install python-dev
-install python-software-properties
+install python3
+install python3-pip
```
Comment (davidhao3300) [inline config/bootstrap.sh]: Remove this line.

File: config/bootstrap.sh [hunk 2]
```diff
@@ -14,12 +14,13 @@ apt-get update
...
+install python3-dev
+install python3-software-properties
+install git
```
Comment (davidhao3300) [inline config/bootstrap.sh:21]: Add the following lines here:
wget https://raw.github.com/pypa/pip/master/contrib/get-pip.py
sudo python3 get-pip.py
rm get-pip.py

**Comment by alanhdu** [comment]
@ohEmily. Just an update, we've merged this in and are now using Python 3.

To upgrade, you need to destroy and recreate the vagrant VM (`vagrant destroy` and `vagrant up`). This takes a while. You'll also need to delete the whoosh-indices (`rm -rf whoosh_index/` and `rm events.db`).

To actually run things, remember to use `python3` (e.g. `python3 run.py` or `python3 create.py`).

Sorry for the headache.

**Pull request merged**